packages/tentales/src/middleware/default-hooks.ts

```diff
@@ -1,24 +1,49 @@
-import { Hook } from "tentales"
+import { Hook, Config } from "tentales"
+import { Log } from "tentales-log"
+import { path } from "ramda"
 import { bodyParserMiddleware } from "./middlewares/body-parser-middleware"
 import { renderMiddleware } from "./middlewares/render-middleware"
 import { fourOhFourMiddleware } from "./middlewares/four-oh-four-middleware"
 import { errorMiddleware } from "./middlewares/error-middleware"
 import { protectServiceRoutesMiddleware } from "./middlewares/protect-service-routes-middelware"
 
-const DEFAULT_HOOKS_PUBLIC: Hook[] = [
-  ["errorMiddleware", [errorMiddleware]],
-  ["bodyParserMiddleware", [bodyParserMiddleware]],
-  ["renderMiddleware", [renderMiddleware]],
-  ["protectServiceRoutesMiddleware", [protectServiceRoutesMiddleware]],
-  ["fourOhFourMiddleware", [fourOhFourMiddleware]],
-]
+const DEFAULT_HOOKS_PUBLIC = (log: Log, config: Config): Hook[] => {
+  const middlewares: Hook[] = [
+    ["errorMiddleware", [errorMiddleware]],
+    ["bodyParserMiddleware", [bodyParserMiddleware]],
+    ["renderMiddleware", [renderMiddleware]],
+  ]
+
+  if (path(["auth", "serverSecret"], config)) {
+    middlewares.push([
+      "protectServiceRoutesMiddleware",
+      [protectServiceRoutesMiddleware],
+    ])
+  } else {
+    log.warn(
+      "Authorization secret not set; no authorization will be carried out.",
```

Comment:
Maybe be explicit and say "... no server-to-server auth..."

Footnote: I've explicitly said "Auth" and not `authorization` and `authentication` in the code. In this case, I guess what we're doing is _authentication_ - checking that the caller really is who s/he says s/he is.

Nice usage, of semicolon!

Reply:
Hmmm, wouldn't this also disable "client-to-server" auth? Maybe something like "this will disable authentication checks for all services".?

Reply:
Hmm.. yeah, it disables client-to-server auth (or rather, it _enables_ a client to talk straight to service). So disabling service-to-service auth makes it open to the public.